main/mio.c

```diff
@@ -990,7 +990,10 @@ char *mio_gets (MIO *mio, char *s, size_t size)
 				mio->impl.mem.pos++;
 				if (s[i] == '\n')
 				{
-					i++;
+					if (i > 0 && s[i-1] == '\r')
+						s[i-1] = '\n';
+					else
+						i++;
```

Comment:
> The problem is that for CR-only line endings the whole file will be treated as a single line. It's true there shouldn't be much of them these days so maybe it's alright-ish but not correct.

BTW, CTags currently does read `\r` line endings as plain characters everywhere `macintosh` isn't defined, so not handling CR has current CTags behavior on most situations.

I don't *really* think it's fully sensible behavior as it seems inconsistent with many other tools (including GCC), but it's how it is since f158c872a72d9ba5042258abefdc07a47e797c5c (as a result of https://sourceforge.net/p/ctags/bugs/1773926/).
Also, how actually common are CR-only files anyway?